allow image attributes in markdown

vivganes opened the pull request.

vivganes committed:
allow attributes for images in markdown

vivganes committed:
allow attributes for images in markdown

vitorpamplona committed:
Forces Coil to request the OriginalSize so that Images can be rendered in bigger sizes than just the line height.

halilozercan committed:
Update dependencies and release new version

This commit updates Compose Multiplatform to 1.4.1
AGP to 7.4.0
Kotlin to 18.10

halilozercan committed:
Fix dokka

vitorpamplona commented:
+1 for this PR. But we might need to check if the composer doesn't need extra info for all possible attributes.

vitorpamplona committed:
BugFix: Image goes over the text (#117)

halilozercan committed:
0.20.0 Update

Version updates;
- Kotlin 1.9.20
- AGP 8.2.0
- Gradle 8.2
- Compose Multiplatform: 1.5.11
- Compose Compiler: 1.5.4

Api Updates;
- Renamed RichText to BasicRichText
- Renamed MaterialRichText to RichText
- Renamed Material3RichText to RichText
- Removed SetupMaterialRichText
- Removed SetupMaterial3RichText

halilozercan committed:
Fix publish workflow

atulgpt committed:
Pass `astNodeType.startNumber` as `startIndex`

This fixes the start number of the markdown list. Markdown can
start with any number except -ve numbers. Rest numbers are
discarded

halilozercan committed:
Introduce richtext-markdown as a generic Markdown renderer

Markdown unfortunately does not have a universally agreed standard. Different flavors, libraries can parse Markdown content in their own ways. `richtext-commonmark` uses `Commonmark` spec and parser to convert any given markdown String into an intermediate Abstract Syntax Tree that has been defined internally since its release.

The idea has always been to extract this ASTNode structure out of this module to enable custom parsing logic to be supplied from outside, and make `richtext-commonmark` only responsible for rendering the given tree using `richtext-ui`.

This change introduces a new module named `richtext-markdown` which is actually just a rename of the existing `richtext-commonmark`. This new module also exposes its internal `AstNode` APIs to enable other modules to render their own AstNode structures.

`richtext-commonmark` becomes a helper module which can parse any given string into an AstNode via `CommonmarkAstNodeParser`. Also, it still provides a `RichTextScope.Markdown` function that simply does the conversion asynchronously, then calls the `Markdown(AstNode)` flavor from `richtext-markdown`.

halilozercan committed:
Version bump 1.5.12

This change includes bunch of misc tech debt cleaning.

- Use Compose Multiplatform dependencies instead of Google's Compose artifacts
- Removed Previews from richtext-ui.
- Renamed CodeBlock platform files.
- Kotlin updated to 1.9.22
- Compose compiler update to 1.5.8
- Changed MarkdownSample to use CommonmarkAstNodeParser and do synchronous Markdown rendering.

halilozercan committed:
Moved LinkClickHandler from Markdown modules to richtext-ui

Text formatting objects should not carry any lambdas for mainly serialization and also control of link clicks should be on a click handler specified at Composable level.

Markdown will take advantage of richtext-ui built-in link click handling.

halilozercan committed:
Add Lazy Markdown Sample

New API allows apps to render long Markdown documents lazily at the top level.

halilozercan committed:
Revert LineHeight bug fix

InlineContent is not supposed to have its own ParagraphStyle. That kind of workaround makes the content essentially "non-inline". Each inline content composable gets its own Paragraph. Instead, we should address the root of the problem which stems from the fact that default Material3 theming adds LineHeight by default.

Sample project now uses Material3 and overrides the default LineHeight to `TextUnit.Unspecified` in its theming. Inline content strictly respecting the line height argument should be addressed by the Compose team.

halilozercan committed:
Fix #96

When rendering markdown lists, some list items may return empty an empty child set. This can be due to markdown content like the following

```
* hello
*
* world
```

In this case, markdown renderer should just put an empty Text in the second item so that `PrefixListLayout` has something to place and report its size correctly.

halilozercan committed:
Fix #109

Lists should use relative placing to show correctly in RTL layouts.

halilozercan committed:
Update Compose version to 1.6.0

This update also prepares the library for 1.0.0-alpha01 release.

halilozercan committed:
Add Spanned to HTML converter

This drops using AndroidView in HtmlBlock.

halilozercan committed:
Update docs

Fix a problem with new module docs titles
Also change the branch name for updating docs

vivganes commented:
@halilozercan do you have any thoughts on this PR?

vivganes committed:
Merge branch 'main' into main

vivganes commented:
@halilozercan  - I have updated this branch with the latest changes.
